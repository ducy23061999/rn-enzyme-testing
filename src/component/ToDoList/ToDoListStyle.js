import {StyleSheet} from 'react-native'
export default StyleSheet.create({
    container: {

    }, 
    itemContainer: {
       margin: 10,
       padding: 10,
       borderRadius: 8 ,
       backgroundColor: '#FF6348',
    },
    itemText: {
        fontSize: 16,
        color: 'white',
        flex: 9
    },
    checkboxContainer: {
       alignSelf: 'flex-end',
       flex: 1,
    },
    checkbox: {
        width: 50,
        height: 50
    }
})