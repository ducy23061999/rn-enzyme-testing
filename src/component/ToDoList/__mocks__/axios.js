
const dumpList = [
    {
        id: 1,
        name: 'do laudary',
        isCheck: false
    }, 
    {
        id: 2,
        name: 'do homework',
        isCheck: true
    },
    {
        id: 3,
        name: 'have dinner',
        isCheck: false
    },
    {
        id: 4,
        name: 'gather with the gang',
        isCheck: false
    }
]

module.exports = {
    get: (url) => {
        return Promise.resolve({
            data: dumpList
        })
    },
    dumpList
}