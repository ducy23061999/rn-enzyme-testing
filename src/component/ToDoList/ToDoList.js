import React, {useEffect, useState} from 'react'
import {View, FlatList, Text, TouchableOpacity} from 'react-native'
import styles from './ToDoListStyle'
import layout from '../../layout'
import CheckBox from '@react-native-community/checkbox'
import _ from 'lodash'
import axios from 'axios'

export default function ({}) {

    const URL = "https://8000-a51b5ad7-0f0d-4c4d-aa72-b9249f88d170.ws-us02.gitpod.io/data"

    const [todos, setToDos] = useState([])
    useEffect(() => {
        axios.get(URL)
        .then(todos => {
            const todoList = JSON.parse(todos.data)
            setToDos(todoList)
        })
        .catch(e => setToDos([]));
    }, [])

    const actions = {
        onItemClick: (item) => {
           const newTodos = todos.map(todo => {
               return {
                    id: todo.id,
                    name: todo.name,
                    isCheck: todo.id == item.id ? !todo.isCheck : todo.isCheck
                }
           })

           setToDos(newTodos);
        }

    }

    return (
        <View>
            <FlatList 
                data = {todos}
                keyExtractor = {item => item.id}
                renderItem = {({item}) => (<ListItem item = {item} onPressItem = {actions.onItemClick}/>) }
            />
        </View>
    )
}

export const ListItem = ({item, onPressItem}) => {
    return (
        <TouchableOpacity 
            style = {[layout.rowStart, styles.itemContainer]}
            onPress = {() => onPressItem(item)}
        >
            <Text style = {styles.itemText}>{item.name}</Text>
            <View style = {styles.checkboxContainer}>
                <CheckBox style = {styles.checkbox}
                    value = {item.isCheck}
                />
            </View>
        </TouchableOpacity>
    )
}