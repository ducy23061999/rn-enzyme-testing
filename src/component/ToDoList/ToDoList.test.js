import React, {useEffect} from 'react'
import axios from 'axios';
import {dumpList} from './__mocks__/axios'
import ToDoList, {ListItem} from './ToDoList'
import { mount, render, shallow } from 'enzyme';
import 'jsdom-global/register';
import { FlatList } from 'react-native';

jest.mock('axios');

describe("ToDoList", () => {

    describe("Rendering", () => {
        it("should be render with mock list", () => {
            jest.spyOn(React, 'useEffect').mockImplementation(f => f())
            const spy = jest.spyOn(axios, 'get');
            const toDoList = mount(
                <ToDoList />
            );
            
            const numberOfChild = toDoList.find(ListItem);
            
            expect(spy).toBeCalled();
            expect(spy).toBeCalledTimes(1);
            expect(toDoList).toMatchSnapshot();
            // expect(numberOfChild.length).toEqual(0) //Bug

        })
    })
})