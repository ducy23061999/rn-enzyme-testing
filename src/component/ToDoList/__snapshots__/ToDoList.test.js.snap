// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoList Rendering should be render with mock list 1`] = `
<_default>
  <View>
    <View>
      <FlatList
        data={Array []}
        disableVirtualization={false}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onEndReachedThreshold={2}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        updateCellsBatchingPeriod={50}
        windowSize={21}
      >
        <VirtualizedList
          data={Array []}
          disableVirtualization={false}
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          onEndReachedThreshold={2}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          updateCellsBatchingPeriod={50}
          viewabilityConfigCallbackPairs={Array []}
          windowSize={21}
        >
          <ScrollView
            data={Array []}
            disableVirtualization={false}
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            onContentSizeChange={[Function]}
            onEndReachedThreshold={2}
            onLayout={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            stickyHeaderIndices={Array []}
            updateCellsBatchingPeriod={50}
            viewabilityConfigCallbackPairs={Array []}
            windowSize={21}
          >
            <RCTScrollView
              data={Array []}
              disableVirtualization={false}
              getItem={[Function]}
              getItemCount={[Function]}
              horizontal={false}
              initialNumToRender={10}
              keyExtractor={[Function]}
              maxToRenderPerBatch={10}
              onContentSizeChange={[Function]}
              onEndReachedThreshold={2}
              onLayout={[Function]}
              onMomentumScrollEnd={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              removeClippedSubviews={false}
              renderItem={[Function]}
              scrollEventThrottle={50}
              stickyHeaderIndices={Array []}
              updateCellsBatchingPeriod={50}
              viewabilityConfigCallbackPairs={Array []}
              windowSize={21}
            >
              <RCTScrollView
                data={Array []}
                disableVirtualization={false}
                getItem={[Function]}
                getItemCount={[Function]}
                horizontal={false}
                initialNumToRender={10}
                keyExtractor={[Function]}
                maxToRenderPerBatch={10}
                onContentSizeChange={[Function]}
                onEndReachedThreshold={2}
                onLayout={[Function]}
                onMomentumScrollEnd={[Function]}
                onScroll={[Function]}
                onScrollBeginDrag={[Function]}
                onScrollEndDrag={[Function]}
                removeClippedSubviews={false}
                renderItem={[Function]}
                scrollEventThrottle={50}
                stickyHeaderIndices={Array []}
                updateCellsBatchingPeriod={50}
                viewabilityConfigCallbackPairs={Array []}
                windowSize={21}
              >
                <View>
                  <View />
                </View>
              </RCTScrollView>
            </RCTScrollView>
          </ScrollView>
        </VirtualizedList>
      </FlatList>
    </View>
  </View>
</_default>
`;
